src/Microsoft.Identity.Web/Resource/RegisterValidAudience.cs

```diff
@@ -67,7 +67,8 @@ internal class RegisterValidAudience
                 throw new SecurityTokenValidationException("Token is not JWT token.");
             }
 
-            string validAudience;
+            TokenValidationParameters validationParametersClone = validationParameters.Clone();
```

Comment:
Should this line be removed from here and added in extension methods like you mentioned [here](https://github.com/AzureAD/microsoft-identity-web/issues/241#issuecomment-651828106)?

Reply:
yes, probably, @pmaytak. That's assigned to you, right?

Reply:
@pmaytak i would have to review the call graph, but i would think Clone() would be called higher in the stack.

Reply:
@brentschmaltz This method is the Validator for every request. To call this once, we would put it into `AddMicrosoftWebApi` method where we set up JwtBearer

Reply:
changed. thanks for the feedback